### ISSUE TITLE
GenericContainer with type: airship skips empty lines in stdout

### ISSUE THREAD

**Issue by aodinokov** [airshipit/airshipctl#471]
**Describe the bug**
Before switching back to GenericContainer type:krm (see [this patchset](https://review.opendev.org/c/airship/airshipctl/+/776494)) execution of

``` sh
airshipctl phase run secret-generate
```

during gate script run produced file `/tmp/airship/airshipctl/manifests/site/test-site/target/generator/results/generated/secrets.yaml`
that contains:

```
...
  pgp:
  - created_at: '2021-02-12T17:01:44Z'
    enc: |-
      -----BEGIN PGP MESSAGE-----
      wcBMAyUpShfNkFB/AQgAibVYA6Cu3LcZ0/Q//4DRpUnVQ8iRUfTBAzDihvE36hFt
      haKwbA/zwdivwNpVCdyw0qoAGwMrXlaSFhsrpdXDNV1dPqVoOzRd5EBIl13xbQGP
      hqR4c4BKIkJM4hGO3LpNNLi6cR9lMmUi06TGVp2GkO8aCVmbTK6Q8RdHRtKisxfb
      pEpiMl9vpequ2IgnWhd+XSy6rCMWpldLzqT1dBMSjSON0TBtLOXB2gqWaszGNhDs
      pfuYo1F0xO86HblgOURTLJ+lr0rhPMn55iiNL1JG5hQcj0to4UKTCKCpOZZrAk0n
      MbfrwIRDC9Nd5xVjl/TNA1IQN9DAapHYWMMHsl3LOdLgAeR0uJnuZ5rwHEokMits
      zhxV4RHG4BfgjOHP5uBw4ubrJxjgHeXL4YBiGY+oPNtaLRI+xzXtw7uT27pwa0ww
      7PFJ3SGcZeCV5ImYQXqdwF70smk47EMHNUbijA5WNeEpkQA=
      =Zv9X
      -----END PGP MESSAGE-----
    fp: FBC7B9E2A4F9289AC0C1D4843D16CEE4A27381B4
```

The expected result should look like

```
...
  pgp:
  - created_at: '2021-02-12T17:01:44Z'
    enc: |-
      -----BEGIN PGP MESSAGE-----

      wcBMAyUpShfNkFB/AQgAibVYA6Cu3LcZ0/Q//4DRpUnVQ8iRUfTBAzDihvE36hFt
      haKwbA/zwdivwNpVCdyw0qoAGwMrXlaSFhsrpdXDNV1dPqVoOzRd5EBIl13xbQGP
      hqR4c4BKIkJM4hGO3LpNNLi6cR9lMmUi06TGVp2GkO8aCVmbTK6Q8RdHRtKisxfb
      pEpiMl9vpequ2IgnWhd+XSy6rCMWpldLzqT1dBMSjSON0TBtLOXB2gqWaszGNhDs
      pfuYo1F0xO86HblgOURTLJ+lr0rhPMn55iiNL1JG5hQcj0to4UKTCKCpOZZrAk0n
      MbfrwIRDC9Nd5xVjl/TNA1IQN9DAapHYWMMHsl3LOdLgAeR0uJnuZ5rwHEokMits
      zhxV4RHG4BfgjOHP5uBw4ubrJxjgHeXL4YBiGY+oPNtaLRI+xzXtw7uT27pwa0ww
      7PFJ3SGcZeCV5ImYQXqdwF70smk47EMHNUbijA5WNeEpkQA=
      =Zv9X
      -----END PGP MESSAGE-----
    fp: FBC7B9E2A4F9289AC0C1D4843D16CEE4A27381B4
```

**Steps To Reproduce**
Minimal steps to reproduce the issue:
1. revert https://review.opendev.org/c/airship/airshipctl/+/776494
2. run gating scripts from tools/gate/ in the correct order.
3. it will fail on the step 23_generate_secrets.sh after generation of secret, because incorrect secret format doesn't allow to decrypt it

**Expected behavior**

**Environment**
- airshipctl Version: all
- Operating System: Ubuntu 18.04 and 20.04
- Kernel version: 4.15.0-135-generic
- Kubernetes Version: n/a
- Go version: 1.15+
- Hardware specs: 4 vCPUs, 32GB RAM

**Comment by mattmceuen** [comment]
@aodinokov I know you put in a workaround for this but which switches back to a KRM format until the Airship format is handled correctly.  With that in mind, what is the urgency on resolving this bug -- do you think it's required for Airship v2.0, or could it wait for a v2.1?

**Comment by teoyaomiqui** [comment]
@mattmceuen, In my opinion, it is not urgent. Airship containers still can be used, as designed to perform all actions, for example, kubectl commands, or image building, while KRM containers are used to process YAMLs.

**Comment by aodinokov** [comment]
@mattmceuen I agree with @teoyaomiqui - KRM-based generic container is enough. Airship-generic container was the attempt to keep the same API, but extend features, e.g. be able to call container in privileged mode, that was necessary for image-builder. Since image-builder was integrated in another way, it's not urgent.

**Comment by teoyaomiqui** [comment]
I don't think it is a good first issue, the fix may not be lengthy, but the design of the solution requires a deep understanding of go, docker SDK, krm functions, and airshipctl itself.
@jezogwza I will remove the `good first issue` label. If you think otherwise, please put it back :).

**Comment by aodinokov** [comment]
> I don't think it is a good first issue, the fix may not be lengthy, but the design of the solution requires a deep understanding of go, docker SDK, krm functions, and airshipctl itself.

I completely agree with @teoyaomiqui. Based on what I saw in it won't be so easy.

**Comment by joshuaherrera** [comment]
I can work on this, please assign to me.

**Comment by aodinokov** [comment]
@joshuaherrera , I don't think it's relevant anymore,
since we copied runfn and plan to have only 1 implementation that can do everything instead of 2: Airship-based and kyaml-based as we had before.

cc @raliev12

**Comment by joshuaherrera** [comment]
@aodinokov ok if this isn't relevant I will work on another issue. Thank you.

**Comment by eak13** [comment]
@aodinokov do we want to go ahead & close this then? Thanks.

**Comment by aodinokov** [comment]
We were supposed to get rid of Airship type of generic containers in https://github.com/airshipit/airshipctl/issues/597
Since it's closed, I guess we can close this as well.

cc: @raliev12
